java/clients/src/main/java/sleeper/clients/deploy/jar/PublishJarsToRepo.java

```diff
@@ -0,0 +1,204 @@
+     */
+    public static void main(String[] args) throws Exception {
+        if (args.length < 3 || args.length > 3) {
+            throw new IllegalArgumentException("Usage: <jars-dir> <repository url> <m2 settings server id>");
+        }
+
+        builder()
+                .pathOfJarsDirectory(Path.of(args[0]))
+                .repoUrl(args[1])
+                .m2SettingsServerId(args[2])
+                .build()
+                .upload();
+    }
+
+    /**
+     * This method loops through the client jars and lambda jars, publishing each one to the supplied maven repository.
+     *
+     * @throws InterruptedException if the current thread was interrupted
+     * @throws IOException          if soemthing goes wrong with running the maven commands
+     */
+    public void upload() throws IOException, InterruptedException {
+        for (ClientJar clientJar : clientJars) {
+            deployJars(clientJar.getFormattedFilename(version), clientJar.getArtifactId());
+        }
+
+        for (LambdaJar lambdaJar : lambdaJars) {
+            deployJars(lambdaJar.getFormattedFilename(version), lambdaJar.getArtifactId());
+        }
+    }
+
+    //Requires matching server with auth details in local m2 settings.xml <servers>
+    private void deployJars(String filename, String artifactId) throws IOException, InterruptedException {
+        commandRunner.run("mvn", "deploy:deploy-file", "-q",
+                "-Durl=" + repoUrl,
+                "-DrepositoryId=" + m2SettingsServerId, //Requires matching server with auth details in local m2 settings.xml <servers>
+                "-Dfile=" + pathOfJarsDirectory.resolve(filename),
+                "-DgroupId=sleeper",
+                "-DartifactId=" + artifactId,
+                "-Dversion=" + version,
+                "-DgeneratePom=false");
+    }
+
+    /**
+     * Builder for publish jars to repo object.
+     */
+    public static final class Builder {
+        private Path pathOfJarsDirectory;
+        private String repoUrl;
+        private String m2SettingsServerId;
+        private String version = SleeperVersion.getVersion();
+        private CommandPipelineRunner commandRunner = CommandUtils::runCommandInheritIO;
+        private List<ClientJar> clientJars = ClientJar.getAll();
+        private List<LambdaJar> lambdaJars = LambdaJar.getAll();
+
+        private Builder() {
+        }
+
+        /**
+         * Sets the path to the jars directory.
+         *
+         * @param  pathOfJarsDirectory the path to the jars folder
+         * @return                     the builder for method chaining
+         */
+        public Builder pathOfJarsDirectory(Path pathOfJarsDirectory) {
+            this.pathOfJarsDirectory = pathOfJarsDirectory;
+            return this;
+        }
+
+        /**
+         * Sets the maven reposiotry url to publish to.
+         *
+         * @param  repoUrl the url of the maven repository to publish to
```

Comment:
Maven should be capitalised as it's a proper noun.

There's a typo at "reposiotry" in the method description.